opslevel/resource_opslevel_property_definition.go

```diff
@@ -1,125 +1,208 @@
+	if planModel.PropertyDisplayStatus.ValueString() != "" {
+		propertyDisplayStatus = opslevel.RefOf(opslevel.PropertyDisplayStatusEnum(planModel.PropertyDisplayStatus.ValueString()))
+	}
+	input := opslevel.PropertyDefinitionInput{
+		AllowedInConfigFiles:  planModel.AllowedInConfigFiles.ValueBoolPointer(),
+		Description:           planModel.Description.ValueStringPointer(),
+		Name:                  planModel.Name.ValueStringPointer(),
+		PropertyDisplayStatus: propertyDisplayStatus,
+		Schema:                definitionSchema,
+	}
+	definition, err := resource.client.CreatePropertyDefinition(input)
+	if err != nil || definition == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to create definition with name '%s', got error: %s", *input.Name, err))
+		return
+	}
+
+	stateModel := NewPropertyDefinitionResourceModel(*definition)
+	stateModel.LastUpdated = timeLastUpdated()
+	tflog.Trace(ctx, fmt.Sprintf("created a definition resource with id '%s'", definition.Id))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (resource *PropertyDefinitionResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var planModel PropertyDefinitionResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	id := planModel.Id.ValueString()
+	definition, err := resource.client.GetPropertyDefinition(id)
+	if err != nil || definition == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to read definition with id '%s', got error: %s", id, err))
+		return
+	}
+
+	stateModel := NewPropertyDefinitionResourceModel(*definition)
+	tflog.Trace(ctx, fmt.Sprintf("read a definition resource with id '%s'", id))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (resource *PropertyDefinitionResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var planModel PropertyDefinitionResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	definitionSchema, err := opslevel.NewJSONSchema(planModel.Schema.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("config error", fmt.Sprintf("unable to use definition schema '%s', got error: %s", planModel.Schema.ValueString(), err))
+		return
+	}
+
+	id := planModel.Id.ValueString()
+	var propertyDisplayStatus *opslevel.PropertyDisplayStatusEnum
+	if planModel.PropertyDisplayStatus.ValueString() != "" {
+		propertyDisplayStatus = opslevel.RefOf(opslevel.PropertyDisplayStatusEnum(planModel.PropertyDisplayStatus.ValueString()))
+	}
+	input := opslevel.PropertyDefinitionInput{
+		AllowedInConfigFiles:  planModel.AllowedInConfigFiles.ValueBoolPointer(),
+		Description:           planModel.Description.ValueStringPointer(),
+		Name:                  planModel.Name.ValueStringPointer(),
```

Comment:
```suggestion
		AllowedInConfigFiles:  opslevel.RefOf(planModel.AllowedInConfigFiles.ValueBool()),
		Description:           opslevel.RefOf(planModel.Description.ValueString()),
		Name:                  opslevel.RefOf(planModel.Name.ValueString()),
```
This will allow unsetting these fields. I'm less sure about the bool one so double check that one

Reply:
@davidbloss Thanks - name and the bool are required so I am going to leave those unchanged but description should be unsettable. Pushed commit for this